components/codeChecker/massPreview/Grid.vue

```diff
@@ -0,0 +1,23 @@
+<template>
+  <div class="grid grid-cols-3 gap-4">
+    <div
+      v-for="(item, index) in items"
+      :key="`${index}_grid_item`"
+      class="relative w-full aspect-square">
+      <NeoSkeleton
+        v-if="item.loading"
+        class="border border-border-color overflow-hidden absolute h-full"
+        width="100%"
+        height="100%"
+        :rounded="false"
+        no-margin />
+
+      <slot :item="item" />
```

Comment:
so, it seems that item  contains more then just loading

isn't it overly complicated?

pass the loading array to this Grid wrapper
and the items directly to BaeMediaItem and to CodeCheckerSandboxIFrame

Reply:
done